script.js

```diff
@@ -1,4 +1,227 @@
+          challenger = challenger + 1;
+        } else if (arrayCombined[0] < arrayCombined[1]) {
+          winningPlayer = challenger;
+          challenger = challenger + 1;
+        }
+      } else if (normalOrReverse == `reverse`) {
+        if (arrayCombined[0] < arrayCombined[1]) {
+          winningPlayer = winningPlayer;
+          challenger = challenger + 1;
+        } else if (arrayCombined[0] > arrayCombined[1]) {
+          winningPlayer = challenger;
+          challenger = challenger + 1;
+        }
+      }
+      playerStage = `numberOfPlayers`;
+      challenger = 2;
+      return (
+        `Player ` +
+        winningPlayer +
+        ` wins this round so he/she won the game!` +
+        `<br>` +
+        `<br>` +
+        `Congrats! 🎉🎊💯 ` +
+        `<br>` +
+        `<br>` +
+        `Enter the number of players for the next game!`
+      );
+    }
+  }
+};
+
+// function to determine stages of game :
+// choose X players -> chooes X dice -> choose game type -> roll dice
+var gameRound = function (input) {
+  var message = ``;
+  if (playerStage == `numberOfPlayers`) {
+    winningPlayer = 1;
+    if (input > 1) {
+      numberOfPlayers = input;
+      playerStage = `choosedice`;
+      message = `${numberOfPlayers} players! <br> Enter number of dice! <br> <br> Playing next: Player ${winningPlayer} & ${challenger}`;
+    } else {
+      message = `Enter number of players (at least two) !`;
+    }
+  } else if (playerStage == `choosedice`) {
+    if (input > 0) {
+      numberOfDice = input;
+      playerStage = `normalreverse`;
+      message = `${numberOfPlayers} players! <br> ${numberOfDice} dice(s). <br> Now choose game mode "normal" or "reverse".<br> <br> Playing next: Player ${winningPlayer} & ${challenger}`;
+    } else {
+      message = `Please enter number of dice (at least one) ! <br> <br> Playing next: Player ${winningPlayer} & ${challenger}`;
+    }
+  } else if (playerStage == `normalreverse`) {
+    message = `${numberOfPlayers} players! <br> ${numberOfDice} dice(s). <br> ${input.toUpperCase()} game! <br> <br> Click submit to roll your dice!<br> <br> Playing next: Player ${winningPlayer} & ${challenger}`;
+    if (input == "normal" || input == "reverse") {
+      normalOrReverse = input;
+      playerStage = `playgame`;
+    } else {
+      message = `Please choose "normal" or "reverse".<br> <br> Playing next: Player ${winningPlayer} & ${challenger}`;
+    }
+  } else if (playerStage == "playgame") {
+    var playgame = rollXDiceXPlayer();
+    var winner = generateCombinedNumWinner();
+    message =
+      normalOrReverse.toUpperCase() +
+      ` round!` +
+      `<br>` +
+      `<br>` +
+      playgame +
+      `<br>` +
+      `<br>` +
+      winner;
+    numberOfDice = ``;
+    arrayCombined = [];
+  }
+  return message;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var game = gameRound(input);
+  var myOutputValue = game;
   return myOutputValue;
 };
```

Comment:
The code quality is really very good. Easy to read and understand. Great work... keep it up!!!